src/components/TopBar.tsx

```diff
@@ -0,0 +1,523 @@
+import fcIconWhite from 'src/images/brands/firecloud/FireCloud-icon-white.svg';
+import headerRightHexes from 'src/images/brands/terra/header-right-hexes.svg';
+import { Ajax } from 'src/libs/ajax';
+import { isBaseline, isBioDataCatalyst, isDatastage, isFirecloud, isTerra } from 'src/libs/brand-utils';
+import colors from 'src/libs/colors';
+import { getConfig } from 'src/libs/config';
+import { withErrorReporting } from 'src/libs/error';
+import { FormLabel } from 'src/libs/forms';
+import { topBarLogo } from 'src/libs/logos';
+import * as Nav from 'src/libs/nav';
+import { useStore } from 'src/libs/react-utils';
+import { authStore, contactUsActive, userStore } from 'src/libs/state';
+import * as Style from 'src/libs/style';
+import * as Utils from 'src/libs/utils';
+
+const topBarStyle: CSSProperties = {
+  flex: 'none',
+  height: Style.topBarHeight,
+  display: 'flex',
+  alignItems: 'center',
+  borderBottom: `2px solid ${colors.primary(0.55)}`,
+  zIndex: 2,
+  boxShadow: '3px 0 13px 0 rgba(0,0,0,0.3)',
+};
+
+const pageTitleStyles: CSSProperties = {
+  color: isTerra() ? 'white' : colors.dark(),
+  fontSize: 22,
+  fontWeight: 500,
+  textTransform: 'uppercase',
+};
+
+const navSectionStyles: CSSProperties = {
+  flex: 'none',
+  height: 70,
+  padding: '0 28px',
+  fontWeight: 600,
+  borderTop: `1px solid ${colors.dark(0.55)}`,
+  color: 'white',
+};
+
+const navBackgroundStyles: CSSProperties = {
+  position: 'absolute',
+  left: 0,
+  right: 0,
+  top: 0,
+  bottom: 0,
+  overflow: 'auto',
+  cursor: 'pointer',
+  zIndex: 2,
+};
+
+const navIconStyles: CSSProperties = {
+  marginRight: 12,
+  flex: 'none',
+};
+
+const navContainerStyles = (state: string): CSSProperties => ({
+  ...(state === 'entered' ? {} : { opacity: 0, transform: 'translate(-2rem)' }),
+  transition: 'opacity 0.2s ease-out, transform 0.2s ease-out',
+  paddingTop: Style.topBarHeight,
+  width: 290,
+  color: 'white',
+  position: 'absolute',
+  cursor: 'default',
+  backgroundColor: colors.dark(0.8),
+  height: '100%',
+  boxShadow: '3px 0 13px 0 rgba(0,0,0,0.3)',
+  zIndex: 2,
+  display: 'flex',
+  flexDirection: 'column',
+});
+
+const NavItem = ({ children, ...props }): ReactNode => (
+  <Clickable
+    {..._.merge(
+      {
+        style: { display: 'flex', alignItems: 'center', color: 'white', outlineOffset: -4 },
+        hover: { backgroundColor: colors.dark(0.55) },
+      },
+      props
+    )}
+  >
+    {children}
+  </Clickable>
+);
+
+const NavSection = ({ children, ...props }): ReactNode => (
+  <div role='listitem'>
+    <NavItem {..._.merge({ style: navSectionStyles }, props)}>{children}</NavItem>
+  </div>
+);
+
+const DropDownSubItem = ({ children, ...props }): ReactNode => (
+  <div role='listitem'>
+    <NavItem {..._.merge({ style: { padding: '0 3rem', height: 40, fontWeight: 500 } }, props)}>{children}</NavItem>
+  </div>
+);
+
+interface DropDownSectionProps extends PropsWithChildren {
```

Comment:
I've been trying to understand when one should use `type` instead of `interface`. In all the places I saw where extend `PropsWIthChildren`, it's done as a type. Is that for any particular reason? And what advise do folks have about when to use type instead of of interface?

Example:

```
type LabeledRadioGroupProps = PropsWithChildren<{
  style?: React.CSSProperties;
}>;

Reply:
An interface defines the shape of an object, but a type is more broad in what it can do.

For example, you can include primatives in a type, define it based on other types, etc. But an interface can only define an object. It can extend another interface, but it can't be say, a union type, or be derived from a type.

Personally, I generally use `interface` if I'm defining a simple concrete object, particularly if I'm exporting it.